### ISSUE TITLE
Integrate `safety`

### ISSUE THREAD

**Issue by altendky** [Julian/mkenv#47]
https://github.com/pyupio/safety

seems like a good thing to be using.  or something like it if there are other options.

**Comment by Julian** [comment]
Yeah these days I tend to generate tox.inis that run that, but mkenv came before I started using it.

Seems reasonable to add.

**Comment by altendky** [comment]
I don't think I was clear and I'm not sure if you got my intent.  I was thinking it would be nice to check against the envs that mkenv makes for installing programs.  So you could `mkenv security` or such to see if any of the envs have known security issues.

But sure also against mkenv itself so we can correct issues quickly. Though GitHub does some automatic checks.

**Comment by Julian** [comment]
Ah OK yeah I definitely didn't assume you meant something like that, although I guess you're coming from https://docs.pipenv.org/#pipenv-check which I had no idea was a thing -- to be honest that seems way out of the boundaries of a tool like this (or pipenv) but maybe I have to think about that more. My initial reaction though would tell me that tools like this should be providing ways to iterate over present virtualenvs, and then you just pass that into some other tool like `safety` that does the check.

I.e., `mkenv list | parallel safety` should be the way to do this.

But yeah who knows maybe it will grow on me.

**Comment by altendky** [comment]
I guess I see mkenv as a thing that's already wrapping virtualenv, why not safety as well.  And we definitely can't recommend using parallel given its lack of windows availability afaik.  It seems it would also be appropriately to (configurably) check after an update to an env.

Certainly I mean to use existing code like safety.  You don't feel good providing an integrated subcommand to trigger it or integrating it (configurably) into other activities?

**Comment by altendky** [comment]
And sure, pipenv made me think of it.  I don't use pipenv because it doesn't work for me (by design at this point and by bugs) but unlike some I don't see it as a totally broken concept.

**Comment by Julian** [comment]
> And we definitely can't recommend using parallel given its lack of windows availability afaik.

`paralell` is just "for loop" -- so, for Windows (or Powershell) whatever the equivalent is there, potentially just literally a for loop.

What I mean to say is I didn't intend for `mkenv` to be "here is the One True Way to Do Python Development as a Whole". safety is a tool that you run to check if some Python environment is safe -- why *should* mkenv manage it other than by providing generic hooks to list what it's managing, unless it is going to dictate, in a very opinionated way, things like: "Everyone should use safety, and flake8, and X and Y and Z", etc. What's wrong with the *existing* way to run safety, combined with the existing way to loop over things? And if mkenv manages safety, should it also manage [readme_renderer](https://pypi.org/project/readme_renderer/) and [ebb-lint](https://pypi.org/project/ebb-lint/) and [sphinx](https://pypi.org/project/sphinx/) and [sphinxcontrib-spelling](https://pypi.org/project/sphinxcontrib-spelling/), all being things I think every project should be using?

(All the above being an elaborated explanation of my initial response, so still happy to be convinced)

> I don't use pipenv because it doesn't work for me (by design at this point and by bugs) but unlike some I don't see it as a totally broken concept.

I agree with all of this sentence :)

**Comment by altendky** [comment]
I'm mostly thinking about mkenv for managing global installs of twine, pipenv, tox, etc.  Basically as a package manager for 'programs' on PyPi.  I think of it helping people not `sudo pip` for those things, nor `pip --user`.  I haven't looked at applying it to my development environment creation yet.  Without it installing itself easily (the bootstrap stuff) and supporting py3 on both Linux and Windows it just isn't something that would be useful for my non-Python coworkers.

But back to safety...  I wouldn't expect to have to pipe the output of an apt command to check for security vulnerabilities.  The same expectation came to mind for mkenv.
